app/queue.py

```diff
@@ -112,6 +121,11 @@ def __move_to_inflight(self, in_flight_key: str, count: int) -> list[str]:
         decoded = [result.decode("utf-8") for result in results]
         return decoded
 
+    def move_from_inflight(self, in_flight_key: str):
+        self.scripts[self.LUA_MOVE_FROM_INFLIGHT](args=[in_flight_key, self._inbox])
+        current_app.logger.warning(f"Moved inflight {in_flight_key} back to inbox {self._inbox}")
+        self._redis_client.delete(in_flight_key)
```

Comment:
The delete as well should be handled by the lua script, else we won't achieve the transaction.